### PR TITLE
[feature] db session contextmanager

### PR DESCRIPTION
**Pull request by icereval**

### CODE REVIEW

**Comment by icereval** [comment]
Needs functional review, basic testing shows a few initial issues.

**Comment by chennan47** [comment]
After cleanup db and login and then choose a project to sync, get this error and after a while app quit itself.
screen shot 2016-02-18 at 9 48 01 am

**Comment by chennan47** [comment]
I still got the error message in the log above, but now it doesn't quit the app. However, sync projects is slow and sometimes freeze.

File: osfoffline/database/__init__.py [hunk 1]
```diff
@@ -30,4 +32,5 @@ def drop_db():
 
 def clear_models():
     for model in CORE_OSFO_MODELS:
-        Session().query(model).delete()
+        with Session() as session:
```
Comment (chrisseto) [inline osfoffline/database/__init__.py]: The for loop should be within the context manager

**Pull request merged**